crates/compiler/solve/src/ability.rs

```diff
@@ -373,7 +380,11 @@ impl ObligationCache {
 
         let ImplKey { opaque, ability } = impl_key;
 
-        let has_declared_impl = abilities_store.has_declared_implementation(opaque, ability);
+        // Every type has the Inspect ability automatically, even opaques with no `implements` declaration.
+        // (Those opaques get a default implementation that returns something along the lines of "<opaque>")
```

Comment:
Is this comment necessary? I think it will get outdated. I would just keep the first one, or none at all.

Reply:
Done